frame/staking-rewards/src/lib.rs

```diff
@@ -375,6 +345,68 @@ pub mod pallet {
 		}
 	}
 
+	impl<T: Config> ManageStaking for Pallet<T> {
+		type AssetId = T::AssetId;
+		type AccountId = T::AccountId;
+		type BlockNumber = <T as frame_system::Config>::BlockNumber;
+		type Balance = T::Balance;
+		type RewardConfigsLimit = T::MaxRewardConfigsPerPool;
+		type StakingDurationPresetsLimit = T::MaxStakingDurationPresets;
+		type RewardPoolId = T::RewardPoolId;
+
+		#[transactional]
+		fn create_staking_pool(
+			pool_config: RewardPoolConfiguration<
+				Self::AccountId,
+				Self::AssetId,
+				Self::BlockNumber,
+				RewardConfigs<Self::AssetId, Self::Balance, Self::RewardConfigsLimit>,
+				StakingDurationToRewardsMultiplierConfig<Self::StakingDurationPresetsLimit>,
+			>,
+		) -> Result<Self::RewardPoolId, DispatchError> {
+			let (owner, pool_id, end_block) = match pool_config {
+				RewardRateBasedIncentive {
+					owner,
+					asset_id,
+					reward_configs: initial_reward_config,
+					end_block,
+					lock,
+				} => {
+					ensure!(
+						end_block > frame_system::Pallet::<T>::current_block_number(),
+						Error::<T>::EndBlockMustBeInTheFuture
+					);
+					let pool_id = RewardPoolCount::<T>::increment()?;
+					let mut rewards = BTreeMap::new();
+					for (_, reward_config) in initial_reward_config.into_iter().enumerate() {
+						rewards.insert(reward_config.0, Reward::from(reward_config.1));
+					}
```

Comment:
I was going to suggest to use iter/collect here, but it looks like this was just moved code?

Reply:
(referring to this while function)

Reply:
Yes, seems can't do much better as anyway need to iterate over reward_configs